### ISSUE TITLE
Customize decision on culling

### ISSUE THREAD

**Issue by rccern** [jupyterhub/jupyterhub-idle-culler#45]
### Proposed change
In our Jupyterhub deployment we have the possibility of assigning users to nodes with GPU, which are a "scarce" resource. For this reason, we would like to have different timeouts depending on which resources the users have booked, e.g. letting sessions expire after 2h since last_activity for nodes with GPUs and 6h for nodes using only CPU.

### Alternative options
Implement this in another culler that we keep for our deployment

### Who would use this feature?
Admins who desire to customize the culler behaviour. Use cases include reserving resources for shorter periods of time for some type of resources, a longer period for users with a specific role...

### (Optional): Suggest a solution
I was thinking about customizing the `should_cull` https://github.com/jupyterhub/jupyterhub-idle-culler/blob/main/jupyterhub_idle_culler/__init__.py#L238  with a configurable callable which takes as inputs the `user` and the `server` objects and returns a boolean. Does it make any sense?

**Comment by rccern** [comment]
I noticed only after publishing that this is essentially a duplicate of #25 , let me know if you prefer that I close this one and continue the discussion there

**Comment by manics** [comment]
https://github.com/jupyterhub/jupyterhub-idle-culler/issues/25 or one of the issues linked there is definitely the best place. Would you mind copying your use case there? Thanks!
